### PR TITLE
Services migration

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/JDG-2116
https://issues.jboss.org/browse/CLOUD-2843

~~@galderz Almost ready, I just have one test failure for `CachingServiceTest::should_put_entries_until_first_one_gets_evicted`~~

~~I'm going to format all the new scripts as well so the format is consistent for the initial import, but I'll do that last.~~

### CODE REVIEW

**Comment by ryanemerson** [comment]
I still need to add the ansible service-broker playbook stuff as well

**Comment by ryanemerson** [comment]
The arquillian tests for the caching service can be run by executing `Make test-ci`, which will deploy everything into openshift and run the functional-tests module.

**Comment by osnipso** [comment]
@ryanemerson :warning: I have rebased this PR on top of the current `datagrid72-dev` branch, so it will cleanly merge with a fast-forward when integrated. However, if you haven't finished working on it yet, you'll have to reset your local branch with `$ git reset --hard <origin_branch>/services-migration` before continuing (you might want to stash any local changes you have or save them in another branch, because this will discard your local `services-migration` branch and replace it with the remote content). Thanks!

**Comment by ryanemerson** [comment]
@osnipso There are likely going to be a few more changes to this PR. In the future, please don't rebase my PRs in this way as it really messes up my local env. If a rebase is required, it is really easy for me to just execute `git pull origin datagrid72-dev --rebase` and force push. Similarly if there are no conflicts, then GH's "Rebase and Merge" option should be sufficient when we want to merge. Finally, if there are conflicts with master then the person issuing the PR should handle these as they know the context of the work and any implications the conflicts may have.

Anyway, I have just rebased my local changes on master and forced pushed to this branch to resolve the issues.

**Comment by osnipso** [comment]
@ryanemerson sure, got it. Just after I did it, I thought you might not have finished the work on the PRs, my bad. Won't do it again.

**Comment by galderz** [comment]
>  Finally, if there are conflicts with master then the person issuing the PR should handle these as they know the context of the work and any implications the conflicts may have.

Indeed, that's the general approach we take in all Infinispan repos. I believe is also what repos out there do.

File: modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   export SASL_SERVER_NAME=${SERVICE_NAME}
+}
+
+# Called automatically as script is executed in openshift-common.sh as part of CONFIGURE_SCRIPTS
+function configure() {
+   # If service profile exists, then configure profile
+   if [ -n ${SERVICE_PROFILE} ]; then
+
+      # Perform profile specific configuration
+      if [ "${SERVICE_PROFILE}" == "caching-service" ]; then
```
Comment (galderz) [inline modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh:35]: @ryanemerson I'm going to have to make XML changes if the service profile is datagrid-service. I thought we agreed those would be done in infinispan-config.sh? Or would we have them here? I wouldn't like to split service profile handling between two scripts and as agreed, I see the benefit of keeping all XML handling in a single place.

**Review by galderz — CHANGES_REQUESTED** [review]
Some comments about service profile handling.

File: modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh (hunk 1 above)
Comment (ryanemerson) [reply modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh:35]: @galderz This is the approach that I was trying to describe in our chat. This script gets executed before `infinispan-config.sh`, so what `infinispan_service_profile.sh` does is set the required env variables for the chosen service. This means that `infinispan-config.sh` is still responsible for all of the XML manipulation, but we get all of the service configuration in one place.

Comment (ryanemerson) [reply modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh:35]: If you need to add new env variables, then this should be done in `infinispan-config.sh` and then they should just be set in `infinispan_service_profile.sh` when required by a service, i.e. `datagrid-service`.

**Comment by galderz** [comment]
One more thing:

Why is `SASL_SERVER_NAME` different from `APPLICATION_NAME`? From a user perspective, I need to connect to the application via Hot Rod (encrypted), so I need to provide the service name, by default `caching-service-app` and also the auth server name, which by default is `caching-service`. So, I somehow need to track two different pieces of information (which might or might not be different). Why not have the auth server name default to the application name? Also, `SASL_SERVER_NAME` is fixed and independent of the application name, is that right?

**Comment by ryanemerson** [comment]
@galderz `SASL_SERVER_NAME = SERVICE_NAME` and `SERVICE_NAME = APPLICATION_NAME`. In the template `APPLICATION_NAME` now defaults to just `caching-service` as I didn't think that `-app` made sense as we're creating a service not an app. 

Maybe we should change the template param from `APPLICATION_NAME` to `SERVICE_NAME` as well?

File: modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh (hunk 1 above)
Comment (galderz) [reply modules/datagrid/72/launch/added/launch/infinispan_service_profiles.sh:35]: Ok, sounds good.

File: services/caching-service.json [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+                           "successThreshold": 1,
+                           "failureThreshold": 5
+                        },
+                        "readinessProbe": {
+                           "exec": {
+                              "command": [
+                                 "/opt/datagrid/bin/readinessProbe.sh"
+                              ]
+                           },
+                           "initialDelaySeconds": 17,
```
Comment (galderz) [inline services/caching-service.json:226]: Any reason why this is 17 and not 15 or 20?

Comment (ryanemerson) [reply services/caching-service.json:226]: Yes apparently it fixes some issues with stability: https://github.com/jboss-container-images/datagrid-7-image/commit/af374d8a10ae463e24fdbc95f1ce04ec61fac1bf

**Review by galderz — APPROVED** [review]

**Pull request merged**